omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java

```diff
@@ -227,39 +228,61 @@ private void addModelAttributes(PageModel model, Patient patient, Section sectio
+
+                    // Last character reveals relationship direction (aIsToB or bIsToA)
+                    char relationshipDirection = types[i].charAt(types[i].length() - 1);
+                    if (relationshipDirection != 'A') {
+                        if (relationshipDirection != 'B') {
+                            throw new APIException("Relationship direction not specified");
+                        }
+                    }
+                    RelationshipType rt = personService.getRelationshipTypeByUuid(relationshipTypeUUID);
+
+                    if (rt != null) {
+                        Person otherPerson = personService.getPersonByUuid(persons[i]);
+
+                        Person personA = relationshipDirection == 'A' ? otherPerson : patient;
+                        Person personB = relationshipDirection == 'B' ? otherPerson : patient;
+                        if (personA != null && personB != null) {
+
+                            relationships = personService.getRelationships(personA, personB, rt);
+                            if (CollectionUtils.isEmpty(relationships)) {
+                                personService.saveRelationship(new Relationship(personA, personB, rt));
+                            } else {
+                                Relationship relationship = relationships.get(0);
+                                relationship.setPersonA(personA);
+                                relationship.setPersonB(personB);
+                                relationship.setRelationshipType(rt);
+                                Relationship newRelationship = personService.saveRelationship(relationship);
+                                if (!CollectionUtils.isEmpty(existingRelationships)) {
+                                    // if this is an existing relationship we make sure we remove it from the list of relationships to be deleted below
+                                    Iterator<Relationship> iterator = existingRelationships.iterator();
+                                    while (iterator.hasNext()) {
+                                        Relationship next = iterator.next();
+                                        if (next.getId().compareTo(newRelationship.getId()) == 0) {
+                                            iterator.remove();
+                                        }
+                                    }
+                                }
+                            }
                         }
                     }
                 }
             }
+
+            if (!CollectionUtils.isEmpty(existingRelationships)) {
+                //assume that any relationship that was not posted to this form it was deleted?
+                for (Relationship existingRelationship : existingRelationships) {
+                    personService.voidRelationship(existingRelationship, "deleted via the Registration widget");
+                }
+            }
```

Comment:
My apologies, as I'm stepping into something I haven't looked at in quite some time, but doesn't the relatioship widget allow you to customize the relationships you display to certain types?  If so, we need to make sure to apply the same logic when deleting (ie, don't delete types that you can't enter/view via the widget)

Reply:
I don't think so, @mogoodrich , it shows all the patient relationships:
https://github.com/openmrs/openmrs-module-registrationapp/blob/master/omod/src/main/webapp/resources/scripts/field/personRelationship.js#L12

This is an old person relationships widget that PIH is not using.

Reply:
And here is the controller:
https://github.com/openmrs/openmrs-module-registrationapp/blob/master/omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java#L223

Reply:
Sounds good to me, thanks @cioan !